frontend/src/components/searchBar/searchBar.tsx

```diff
@@ -0,0 +1,50 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import React, { useState, useEffect, Dispatch, SetStateAction } from 'react';
+import { FaSearch } from 'react-icons/fa';
+import { theme } from '../../theme';
+import {
+  SearchIconContainer,
+  SearchInputContainer,
+  SearchInputField,
+} from './searchBar-styles';
+
+export interface SearchProps {
+  setData: Dispatch<SetStateAction<any>>;
+  data: any;
+  searchKey: string;
+}
+
+const SearchBar = ({ setData, data, searchKey }: SearchProps) => {
+  const [searchTerm, setSearchTerm] = useState('');
+
+  const handleSearchchange = (e: React.FormEvent<HTMLInputElement>) => {
+    setSearchTerm(e.currentTarget.value);
+  };
+
+  useEffect(() => {
+    if (!searchTerm) {
+      setData(data);
+    } else {
+      setData(
+        data.filter((item: any) =>
+          item[searchKey].toLowerCase().includes(searchTerm.toLowerCase())
+        )
```

Comment:
Yes, the component filters data itself, and then returns the filtered data, so that the filtering logic doesn't have to be inside of the page we use searchBar on. But the component can be used to filter any array of objects. 
But I understand what you mean, sometimes you might need to do your own filtering inside of some page and just use the searchBar for input.